### ISSUE TITLE
Zero length packets

### ISSUE THREAD

**Issue by AljoschaMeyer** [dominictarr/pull-box-stream#10]
Does the protocol allow headers with a length field of 0, which are then directly followed by the next header? As long as the value used as the packet MAC isn't zero, they'd be distinguishable from final packets.
Still I'd prefer if they were disallowed, since a well-behaved user would never produce them in the first place (and there should be better solutions if the goal is to obscure packet boundaries).

**Comment by dominictarr** [comment]
yes I agree they should be disallowed. It's not obvious in the code, but libsodium will error if you try to encrypt/decrypt a zero length packet so sending this wouldn't be possible with the current implementation.
to be explicit, we should change https://github.com/dominictarr/pull-box-stream/blob/master/index.js#L126
to just check if the length field is zero.
